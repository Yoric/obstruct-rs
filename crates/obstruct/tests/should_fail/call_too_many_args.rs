#![feature(associated_const_equality)]

fn test_too_many_args() {
    use obstruct_macros::{destruct, call};
    const A: i32 = 15;
    const B: f64 = 17f64;
    const C: &str = "Out of order";


    destruct!{fn test_foo({a: i32, b: f64, c: &'static str}) -> (i32, f64, &'static str) { (a, b, c) }};

    let (a, b, c) = call! { test_foo, { b: B, a: A, c: C, oops: 0}}; // Too many arguments.
    assert_eq!(a, A);
    assert_eq!(b, B);
    assert_eq!(c, C);
}


fn main() {}
