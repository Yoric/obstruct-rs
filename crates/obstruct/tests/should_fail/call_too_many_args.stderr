error[E0308]: mismatched types
  --> tests/should_fail/call_too_many_args.rs:12:21
   |
12 |     let (a, b, c) = call! { test_foo, { b: B, a: A, c: C, oops: 0}}; // Too many arguments.
   |                     ^^^^^^^^--------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                     |       |
   |                     |       arguments to this function are incorrect
   |                     expected a tuple with 3 elements, found one with 4 elements
   |
   = note: expected tuple `(a<i32>, b<f64>, c<&str>)`
              found tuple `(a<i32>, b<f64>, c<&str>, oops<{integer}>)`
note: function defined here
  --> tests/should_fail/call_too_many_args.rs:10:18
   |
10 |     destruct!{fn test_foo({a: i32, b: f64, c: &'static str}) -> (i32, f64, &'static str) { (a, b, c) }};
   |     -------------^^^^^^^^------------------------------------------------------------------------------
   = note: this error originates in the macro `call` (in Nightly builds, run with -Z macro-backtrace for more info)
